src-cpp/rdkafkacpp.h

```diff
@@ -1365,7 +1366,139 @@ class RD_EXPORT MessageTimestamp {
   int64_t timestamp;               /**< Milliseconds since epoch (UTC). */
 };
 
+/**
+ * @brief Headers object
+ *
+ * This object encapsulates the C implementation logic into a C++ object
+ * for use in RdKafka::Messages object.
+ */
+class RD_EXPORT Headers {
+ public:
+  virtual ~Headers() = 0;
+
+  /**
+   * @brief Header object
+   *
+   * This object represents a single Header with key value pair
+   * and an ErrorCode
+   *
+   * @remark dynamic allocation of this object is not supported.
+   *
+   */
+  class Header {
+   public:
+    Header(const std::string& key,
+           const char* value,
+           RdKafka::ErrorCode err = ERR_NO_ERROR):
+    key_(key), err_(err) {
+        value_container_.assign(value);
+    };
+    
+    std::string key() const {
+        return key_;
+    }
+
+    const char* value() const {
+        return value_container_.c_str();
+    }
+
+    RdKafka::ErrorCode err() const {
+        return err_;
+    }
+    
+   private:
+    std::string key_;
+    RdKafka::ErrorCode err_;
+    std::string value_container_;
+    void *operator new(size_t); /* Prevent dynamic allocation */
+  };
+
+  /**
+   * @brief create a new instance of the Headers object
+   */
+  static Headers *create(size_t initial_size = 8, bool free_rd_headers = true);
```

Comment:
Don't use default param values:
https://google.github.io/styleguide/cppguide.html#Default_Arguments

Document the  `@param`s.

I don't think free_rd_headers should be part of the public API